src/GenerateOverlapMeshParallelExe.cpp

```diff
@@ -0,0 +1,137 @@
+///////////////////////////////////////////////////////////////////////////////
+///
+///	\file    GenerateOverlapMeshParallelExe.cpp
+///	\author  Paul Ullrich
+///	\version August 23, 2021
+///
+///	<remarks>
+///		Copyright 2021 Paul Ullrich
+///
+///		This file is distributed as part of the Tempest source code package.
+///		Permission is granted to use, copy, modify and distribute this
+///		source code and its documentation under the terms of the GNU General
+///		Public License.  This software is provided "as is" without express
+///		or implied warranty.
+///	</remarks>
+
+#include "Announce.h"
+#include "CommandLine.h"
+#include "GridElements.h"
+#include "STLStringHelper.h"
+
+#include "TempestRemapAPI.h"
+
+#if defined(TEMPEST_MPIOMP)
+#include <mpi.h>
+#endif
+
+///////////////////////////////////////////////////////////////////////////////
+
+int main(int argc, char** argv) {
+
+#if defined(TEMPEST_MPIOMP)
+	MPI_Init(&argc, &argv);
+	AnnounceOnlyOutputOnRankZero();
+#endif
+
+	// Input mesh A
+	std::string strMeshA;
+
+	// Input mesh B
+	std::string strMeshB;
+
+	// Output mesh file
+	std::string strOverlapMesh;
+
+	// Output format
+	std::string strOutputFormat;
+
+	// Overlap grid generation method
+	std::string strMethod;
+
+	// Overlap grid generation algorithm
+	std::string strAlgorithm;
+
+	// No validation of the meshes
+	bool fNoValidate;
+
+	// Concave elements may be present in mesh A
+	bool fHasConcaveFacesA;
+
+	// Concave elements may be present in mesh B
+	bool fHasConcaveFacesB;
+
+	// Allow for the case of no overlap element found (enabling this may
+	// cause the mesh generator to generate an incomplete overlap mesh)
+	bool fAllowNoOverlap;
+
+	// Run in parallel
+	bool fParallel;
+
+	// Temporary file directory
+	std::string strTempDir;
+
+	// Verbose
+	bool fVerbose;
+
+	// Parse the command line
+	BeginCommandLine()
+		CommandLineString(strMeshA, "a", "");
+		CommandLineString(strMeshB, "b", "");
+		CommandLineString(strOverlapMesh, "out", "overlap.g");
+		CommandLineString(strOutputFormat, "out_format", "netcdf4");
+		CommandLineStringD(strMethod, "method", "fuzzy", "(fuzzy|exact|mixed)");
+		CommandLineStringD(strAlgorithm, "alg", "lint", "(lint)");
+		CommandLineBool(fNoValidate, "novalidate");
+		CommandLineBool(fHasConcaveFacesA, "concavea");
+		CommandLineBool(fHasConcaveFacesB, "concaveb");
+		CommandLineBool(fAllowNoOverlap, "allow_no_overlap");
+		CommandLineString(strTempDir, "tmpdir", "/tmp");
+		CommandLineBool(fVerbose, "verbose");
+
+		ParseCommandLine(argc, argv);
+	EndCommandLine(argv)
+
+	AnnounceBanner();
+
+#if !defined(TEMPEST_MPIOMP)
+	_EXCEPTIONT("TempestRemap was not compiled with MPI: Parallel operation not supported");
```

Comment:
If MPI is unavailable, can it still default to running in serial ?

Reply:
I figure if you're running GneerateOverlapMeshParallel it should only work if parallelism is enabled.

Reply:
Sure that is fair. I guess if someone wants to verify accuracy or look at scalability etc, they would still run on 1 process with MPI enabled anyway, rather than resorting to a MPI disabled build